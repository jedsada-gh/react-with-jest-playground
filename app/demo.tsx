"use client";

import { useEffect, useState } from "react";
import { getUser } from "./service";

export const DemoComponent = () => {
  const [user, setUser] = useState<any>(null);

  const getUserData = async () => {
    const user = await getUser();
    // TODO: logic data transformer
    const _user = transformUserData(user);
    setUser(_user);
  };

  const transformUserData = (user: any) => {
    if (user.title === "delectus aut autem") {
      user.title = "Title Test Demo";
    }
    return user;
  };

  useEffect(() => {
    getUserData();
  }, []);

  return (
    <div>
      <p data-testid="title">Demo</p>
      <p data-testid="title1">Demo1</p>
      <p data-testid="name">{`${user?.title}`}</p>
    </div>
  );
};
