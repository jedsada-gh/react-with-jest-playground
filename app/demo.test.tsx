import { DemoComponent } from "./demo";
import { render, screen, waitFor } from "@testing-library/react";
import Link from "next/link";
import renderer from 'react-test-renderer';

jest.mock("./service", () => ({
  getUser: () => Promise.resolve({ title: "delectus aut autem" }),
}));

it("DemoComponent render", async () => {
  render(<DemoComponent />);
  expect(screen.getByTestId("title")).toHaveTextContent("Demo");
  await waitFor(() => {
    expect(screen.getByTestId("name")).toHaveTextContent("Title Test Demo");
  })
});

it('renders correctly', () => {
  const tree = renderer
    .create(<DemoComponent/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
