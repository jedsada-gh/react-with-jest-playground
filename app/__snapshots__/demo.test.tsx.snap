// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <p
    data-testid="title"
  >
    Demo
  </p>
  <p
    data-testid="name"
  >
    undefined
  </p>
</div>
`;
